report_thesis/src/sections/pyhat_contribution.tex

```diff
@@ -0,0 +1,22 @@
+\section{PyHAT Contribution}\label{sec:pyhat_contribution}
+As part of our research, we have made several contributions to \gls{pyhat}, which offers a user-friendly interface designed for performing machine learning and data analysis tasks specifically for hyperspectral data.
+This collaboration was initiated through a series of discussions with two members \gls{usgs} responsible for managing \gls{pyhat}, wherein we identified mutual challenges and opportunities for integrating our solutions into the library.
+
+The largest contribution involved the integration of an automatic outlier detection method into the library's \gls{pls} implementation. 
+This method calculates the Mahalanobis distance for each data point and uses the chi-squared distribution to establish a threshold.
+Any datapoint exceeding this threshold is considered an out and removed from the dataset.
+Utilizing two intermediary \gls{pls} models, one as a reference and the other to evaluate the impact of outlier removal, the method iteratively identifies and eliminates outliers while assessing the performance of the second model against the reference model. 
+If the second model demonstrates improved performance compared to the reference model, it replaces the reference model, and the process continues until no further significant improvement is detected. 
```

Comment:
Oh, you're referring to the `best_or_model` variable? Yeah, that's not something we did for PyHAT. That was in our own project for logging.